### PR TITLE
Feature/macos setup support

### PR DESCRIPTION
**Pull request by Cheyans**

### CODE REVIEW

**Comment by Cheyans** [comment]
Don't merge this, I need to fix something.

**Comment by Cheyans** [comment]
Should be good to merge

File: setup_db.sh [hunk 1]
```diff
@@ -1,38 +1,52 @@
...
-echo "Waiting for mysql container..."
-DB_CONTAINER=$(docker run -d --name faf-db -e MYSQL_ROOT_PASSWORD=banana -p 3306:3306 faf-db)
-until nc -z $(docker inspect --format='{{.NetworkSettings.IPAddress}}' ${DB_CONTAINER}) 3306
+# Run faf-db image in a new container and waits until it initialized
+echo "Creating database faf_test..."
+until docker exec -i ${DB_CONTAINER} mysql -uroot -pbanana -e "create database faf_test;" 2>/dev/null;
 do
-  sleep 1
+  echo "Mysql still initializing, retrying in 10 seconds";
+  sleep 10;
```
Comment (micheljung) [inline setup_db.sh]: Since it's in a loop anyway, I'd prefer a shorter timeout :) Usually takes 1-2s or so on my machine

File: README.md [hunk 2]
```diff
@@ -12,6 +12,8 @@ Install [docker](http://docker.com).
 
 Now your FAF database is up and running and contains some dummy data (from `db-data.sql`)
 
+If you want the database schema you can add the `-c` flag, this will dump the schema to `/tmp/dump.sql` on your host system.
+
```
Comment (micheljung) [inline README.md]: How about letting the user specify where to dump?

**Comment by Cheyans** [comment]
Added support for custom schema dump location and a fancy loading spinner

File: setup_db.sh [hunk 3]
```diff
@@ -1,38 +1,74 @@
...
+}
+
+show_help() {
+    printf "Usage: ./setup_db.sh [options]
+       ./setup_db.sh -c /tmp/dump.sql
+       ./setup_db.sh -c /tmp
+
+Options:
+    -d                   Dump db schema to container stdout
+    -c \033[4mfile\033[0m              Dump db schema to provided host location, if directory provided, file name will be dump.sql
```
Comment (micheljung) [inline setup_db.sh]: Is this supposed to look this way?
I think the wording should be `Dump DB schema to provided file location. If a directory is provided, the file name will be dump.sql`

File: README.md (hunk 2 above)
Comment (micheljung) [inline README.md]: Code was updated, but readme wasn't.

File: setup_db.sh (hunk 3 above)
Comment (Cheyans) [inline setup_db.sh]: Yea, it underlines the file, I could add a function to do it, but I think it's fine for now.

**Pull request merged**